crates/llvm-context/src/polkavm/evm/ether_gas.rs

```diff
@@ -40,22 +40,26 @@ pub fn balance<'ctx, D>(
 where
     D: Dependency + Clone,
 {
-    let balance_pointer = context.build_alloca(context.word_type(), "balance_pointer");
-    let address_pointer = context.build_alloca(context.word_type(), "address_pointer");
-    context.build_store(address_pointer, address)?;
+    let address_type = context.integer_type(revive_common::BIT_LENGTH_ETH_ADDRESS);
+    let address_pointer = context.build_alloca_at_entry(address_type, "address_pointer");
+    let address_truncated =
+        context
+            .builder()
+            .build_int_truncate(address, address_type, "address_truncated")?;
+    let address_swapped = context.build_byte_swap(address_truncated.into())?;
+    context.build_store(address_pointer, address_swapped)?;
```

Comment:
I see we are starting to need this in more and more places. It would be great to have a method `fn build_address_argument_store` on the context that does exactly this, returning the address `address_pointer`.

Now this isn't your fault; it is fine like that. I'll leave it up to you to decide if you want to implement the helper on the context. If you do all other occurrences should be changed to use the new helper instead.

Reply:
Updated

Reply:
Good stuff, thanks!